com.unity.shadergraph/Editor/Importers/ShaderSubGraphImporter.cs

```diff
@@ -238,7 +239,54 @@ static void ProcessSubGraph(SubGraphAsset asset, GraphData graph)
             var collector = new PropertyCollector();
             foreach (var node in nodes)
             {
+                int previousPropertyCount = Math.Max(0, collector.properties.Count-1);
+
                 node.CollectShaderProperties(collector, GenerationMode.ForReals);
+
+                // This is a stop-gap to prevent the autogenerated values from JsonObject and ShaderInput from
+                // resulting in non-deterministic import data. While we should move to local ids in the future,
+                // this will prevent cascading shader recompilations.
+                for (int i = previousPropertyCount; i < collector.properties.Count; ++i)
+                {
+                    // Generate a deterministic guid using namespace guids: RFC 4122 §4.3 version 5.
+                    byte[] GenerateNamespaceUUID(string Namespace, string Name)
+                    {
+                        void FlipByNetworkOrder(byte[] bytes)
+                        { bytes = new byte[] { bytes[3], bytes[2], bytes[1], bytes[0], bytes[5], bytes[4], bytes[7], bytes[6] }; }
+
+                        var namespaceBytes = Encoding.UTF8.GetBytes(Namespace);
+                        FlipByNetworkOrder(namespaceBytes);
+                        var nameBytes = Encoding.UTF8.GetBytes(Name);
+                        var hash = SHA1.Create().ComputeHash(namespaceBytes.Concat(nameBytes).ToArray());
+                        byte[] newguid = new byte[16];
+                        Array.Copy(hash, newguid, 16);
+                        newguid[6] = (byte)((newguid[6] & 0x0F) | 0x80);
+                        newguid[8] = (byte)((newguid[8] & 0x3F) | 0x80);
+                        FlipByNetworkOrder(newguid);
+                        return newguid;
+                    }
+
+                    var prop = collector.properties[i];
+                    var namespaceId = node.objectId;
+                    var nameId = prop.referenceName;
+
+                    // If the node's object Id is not actually a guid, use the assetGuid as a namespace instead.
+                    if (!Guid.TryParse(namespaceId, out _))
+                    {
+                        Debug.LogWarning("Node objectId is not a valid Guid, using assetGuid instead." + node.objectId);
```

Comment:
Has this happened in practice?
Non-user-actionable log messages tend to just clutter up the log window; should either make it more actionable / describe some way for the user to fix it, or make it an dev-only message (if not expected?), or remove it.